frontend-react/docs/rs-react-testing-network-calls.md

```diff
@@ -38,77 +39,116 @@ const handlers = [
     /* Successfully returns array of Reports */
     rest.get(`http://localhost:3000${listEndpoint.url}`, (req, res, ctx) => {
         return res(ctx.json(HistoryApi.testResponse(3)));
-    })
-]
+    }),
+];
 ```
 
 The parameters necessary are an endpoint (as a string), and a callback function, inside of which you return your mock data. For more information on what can be done using request (req), response (res), and context (ctx), [visit the msw documentation page](https://mswjs.io/docs/getting-started/mocks)!
 
 Here is an example of returning a failed API response:
 
 ```typescript
-    /* Fails to return array of Reports with 404 */
-    rest.get(`http://localhost:3000${detailFail.url}`, (req, res, ctx) => {
-        return res(ctx.status(404));
-    })
+/* Fails to return array of Reports with 404 */
+rest.get(`http://localhost:3000${detailFail.url}`, (req, res, ctx) => {
+    return res(ctx.status(404));
+});
 ```
+
 > NOTE: As a best practice, I propose we adopt a Success Service / Fail Service pattern, rather than combining all into a single mock service. This isn't necessary for API class tests, but you'll see once we get to components, it really helps simplify things.
-> 
+>
 > See below for more.
 
+For an example see our [lookup table mock server](https://github.com/CDCgov/prime-reportstream/blob/58599e355ad2a0b357005190c185a8b8497f0b81/frontend-react/src/__mocks__/LookupTableMockServer.ts).
+
+### Testing for failure
+
+For the sake of completeness, we need both success and failure tests, which means we're going to need to create two separate mock services in our mock service module. Here's an example of how we can mock failure and success responses for the same request:
+
+```typescript
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+
+const successHandlers = [
+    rest.get(tableListUrl, (req, res, ctx) => {
+        return res(ctx.status(200));
+    }),
+];
+
+const failureHandlers = [
+    rest.get(tableListUrl, (req, res, ctx) => {
+        return res(ctx.status(404));
+    }),
+];
+
+export const successServer = setupServer(...successHandlers);
```

Comment:
You're able to conditionally return within the handler functions. Since each of these requires its own startup/teardown when testing, should we keep them combined as much as possible?

Reply:
Definitely. This documentation is present in the existing doc. https://github.com/CDCgov/prime-reportstream/pull/6648/files#diff-0b4dc3618e3953b315ccddbf2b2536d54903bde6c8869a4d9a4b68d0fda228d6R147 It doesn't seem like all that common of a pattern, but if you do want to test different behaviors from exactly the same request within a given test file, it's an interesting approach. I covered the distinction between this approach and the approach of conditional handling within the callback here https://github.com/CDCgov/prime-reportstream/pull/6648/files#diff-0b4dc3618e3953b315ccddbf2b2536d54903bde6c8869a4d9a4b68d0fda228d6R89 but I could do a separate example if you think that's useful. I generally erred on the side of reducing churn on this document while making sure it was up to date.